examples/solvers/chambolle_pock_tomography_tgv.py

```diff
@@ -2,115 +2,152 @@
 TGV_2 is the second order total generalized variation of ``x``, defined as
 
-    TGV_2(x) = min_y lam_1 ||grad(x) - y||_1 + lam_2 ||eps(y)||_1
+    TGV_2(x) = min_y lam_1 ||Gx - y||_1 + lam_2 ||Ey||_1
 
-where ``grad`` is the (vectorial) spatial gradient and ``eps`` is the matrix
-valued spatial derivative of the vector valued function ``y``.
+where ``G`` is the spatial gradient operating on the scalar-field ``x`` and
+``E`` is the symmetrized gradient operating on the vector-field ``y``.
+Both one-norms ||.||_1 take the one-norm globally and the two-norm locally,
+i.e. ||y||_1 := sum_i sqrt(sum_j y_i(j)^2) where y_i(j) is the j-th value
+of the vector y_i at location i.
 
 The problem is rewritten as
 
-    min_{x, y} ||A(x) - g||_2^2 +  lam_1 ||grad(x) - y||_1 + lam_2 ||eps(y)||_1
+    min_{x, y} ||Ax - g||_2^2 +  lam_1 ||Gx - y||_1 + lam_2 ||Ey||_1
 
 which can then be solved with the Chambolle-Pock method.
+
+References
+----------
+[1] K. Bredies and M. Holler. A TGV-based framework for variational image
+decompression, zooming and reconstruction. Part II: Numerics. SIAM Journal
+on Imaging Sciences, 8(4):2851-2886, 2015.
 """
 
 import numpy as np
 import odl
 
-
+# %%
 # --- Set up the forward operator (ray transform) --- #
 
-
 # Reconstruction space: discretized functions on the rectangle
-# [-20, 20]^2 with 300 samples per dimension.
-reco_space = odl.uniform_discr(
-    min_pt=[-20, -20], max_pt=[20, 20], shape=[300, 300], dtype='float32')
+# [-20, 20]^2 with 100 samples per dimension.
+n = 100
+U = odl.uniform_discr(
+    min_pt=[-20, -20], max_pt=[20, 20], shape=[n, n], dtype='float32')
 
 # Make a parallel beam geometry with flat detector
-geometry = odl.tomo.parallel_beam_geometry(reco_space)
+geometry = odl.tomo.parallel_beam_geometry(U)
 
 # Create the forward operator
-ray_trafo = odl.tomo.RayTransform(reco_space, geometry)
-
+A = odl.tomo.RayTransform(U, geometry)
 
+# %%
 # --- Generate artificial data --- #
 
-
 # Create phantom
-phantom = odl.phantom.shepp_logan(reco_space, modified=True)
+#phantom = odl.phantom.shepp_logan(U, modified=True)
+
+m = 25
+
+e1 = np.linspace(0, 1, n)
+e2 = np.ones([n, 1])
+x = np.outer(e1, e2)
+
+e1 = np.linspace(1, 0, n-2*m)
+e2 = np.ones([n-2*m, 1])
+x[m:-m, m:-m] = np.outer(e1, e2)
+
+phantom = U.element(x)
+
+phantom.show(title='Phantom')
 
 # Create sinogram of forward projected phantom with noise
-data = ray_trafo(phantom)
-data += odl.phantom.white_noise(ray_trafo.range) * np.mean(data) * 0.1
+data = A(phantom)
+data += odl.phantom.white_noise(A.range) * np.mean(data) * 0.1
 
+data.show(title='Simulated data (Sinogram)')
 
+# %%
 # --- Set up the inverse problem --- #
 
-
 # Initialize gradient operator
-gradient = odl.Gradient(reco_space)
+G = odl.Gradient(U, method='forward', pad_mode='symmetric')
+V = G.range
+
+Dx = odl.discr.diff_ops.PartialDerivative(U, 0, method='backward',
+                                          pad_mode='symmetric')
```

Comment:
What the boundary conditions are supposed to do, are to extend the image constantly such that the gradient becomes zero. Am I achieving this with this ODL code?